orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileCacheManagerImpl.java

```diff
@@ -155,11 +171,10 @@ public void put(OrcidProfile orcidProfile) {
 
     public void put(String orcid, OrcidProfile orcidProfile) {
         try {
-            synchronized (profileLockers.obtainLock(orcid)) {
-                profileCache.put(new Element(new OrcidCacheKey(orcid, releaseName), orcidProfile));
-            }
+            profileCache.acquireWriteLockOnKey(orcid);
```

Comment:
hum.....if we are replacing `profileLockers.obtainLock(orcid)) ` then no. If we are saying the original code is incorrect then yes.